dashboard/src/components/DeploymentForm/DeploymentForm.tsx

```diff
@@ -76,6 +90,17 @@ export default function DeploymentForm() {
     return () => {};
   }, [dispatch, packageReference, packageVersion]);
 
+  useEffect(() => {
+    // Populate the service account list if the plugin requires it
+    if (getPluginsRequiringSA().includes(pluginObj.name)) {
+      // We assume the user has enough permissions to do that. Fallback to a simple input maybe?
+      Kube.getServiceAccountNames(targetCluster, targetNamespace).then(saList =>
+        setServiceAccountList(saList.serviceaccountNames),
```

Comment:
I'm storing the SA list in the component's state solely; as it is not shared by other components.

Reply:
Yes, and we need to fetch it whenever the target namespace changes (as you've indicated below with the dependencies), so yep, component state sounds perfect.